main/src/main/scala/sbt/internal/server/BspCompileProgress.scala

```diff
@@ -0,0 +1,31 @@
+/*
+ * sbt
+ * Copyright 2011 - 2018, Lightbend, Inc.
+ * Copyright 2008 - 2010, Mark Harrah
+ * Licensed under Apache License 2.0 (see LICENSE)
+ */
+
+package sbt.internal.server
+
+import xsbti.compile.CompileProgress
+
+private[sbt] final class BspCompileProgress(
+    task: BspCompileTask,
+    underlying: Option[CompileProgress]
+) extends CompileProgress {
+  override def advance(
+      current: Int,
+      total: Int,
+      prevPhase: String,
+      nextPhase: String
+  ): Boolean = {
+    val percentage = current * 100 / total
+    // Report percentages every 5% increments
+    val shouldReportPercentage = percentage % 5 == 0
+    if (shouldReportPercentage) {
+      task.notifyProgress(percentage, total)
+    }
+    underlying.foreach(_.advance(current, total, prevPhase, nextPhase))
+    true
```

Comment:
Now I realize my previous suggestion was a bit too simple. Because we should better return the underlying result if defined:

```suggestion
    underlying.map(_.advance(current, total, prevPhase, nextPhase))
      .getOrElse(true)
```

Also the other methods should redirect to the `underlying` progress. That was kind of what you did before. Sorry for the back and forth.
```scala
override def startUnit(phase: String, unitPath: String): Unit = {
  underlying.foreach(_.startUnit(phase, unitPath))
}

override def afterEarlyOutput(success: Boolean): Unit = {
  underlying.foreach(_.afterEarlyOutput(sucess))
}
```

Reply:
No worries at all! I ended up using `.fold(true)(_.advance(...))` because IntelliJ was trying to suggest I use `.forall` instead of map/getOrElse 😬

Thanks again for your comments!